apps/homescreen/js/app.js

```diff
@@ -395,24 +427,33 @@ const SETTINGS_VERSION = 0;
       return container;
     },
 
-    addAppIcon: function(appOrBookmark, entryPoint) {
-      var id;
+    getIconId: function(appOrBookmark, entryPoint) {
       if (appOrBookmark.id) {
-        id = appOrBookmark.id;
+        return appOrBookmark.id;
       } else {
-        id = appOrBookmark.manifestURL + '/' + (entryPoint ? entryPoint : '');
+        return appOrBookmark.manifestURL + '/' + (entryPoint ? entryPoint : '');
       }
+    },
 
-      var entry = this.startupMetadata.findIndex((element) => {
-        return element.id === id;
-      });
-      var container = this.addIconContainer(entry);
+    addAppIcon: function(appOrBookmark, entryPoint) {
+      var id = this.getIconId(appOrBookmark, entryPoint);
+      var entry = -1;
+
+      if (this.startupMetadata !== null) {
+        entry = this.startupMetadata.findIndex((element) => {
+          return element.id === id;
+        });
+        if (entry === -1) {
+          this.pendingIcons[id] = Array.slice(arguments);
```

Comment:
Was about to suggest spread operator here (i.e. `this.pendingIcons[id] = [...arguments];`), but Firefox is not up to spec on arguments iterator :-(